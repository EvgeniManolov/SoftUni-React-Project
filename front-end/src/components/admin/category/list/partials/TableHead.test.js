import React from 'react';
import TableHead from './TableHead';
import {shallow} from 'enzyme';

it('renders TableHead component', () => {
    const wrapper = shallow(<TableHead/>)
    expect(wrapper).toBeTruthy();
});

it('check for html changes', () => {
  const wrapper = shallow(<TableHead/>)
  expect(wrapper).toMatchSnapshot();
});
