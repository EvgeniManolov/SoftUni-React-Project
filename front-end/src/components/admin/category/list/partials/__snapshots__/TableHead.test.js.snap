// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check for html changes 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TableHead />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <tr>
        <th>
                <SortButtons
                        changeClass={undefined}
                        sort={undefined}
                        sortProperty="name"
                />
        </th>
        <th
                className="text-center"
        />
</tr>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <th>
            <SortButtons
                        changeClass={undefined}
                        sort={undefined}
                        sortProperty="name"
            />
</th>,
          <th
            className="text-center"
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <SortButtons
              changeClass={undefined}
              sort={undefined}
              sortProperty="name"
/>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "changeClass": undefined,
              "sort": undefined,
              "sortProperty": "name",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "th",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "text-center",
          },
          "ref": null,
          "rendered": null,
          "type": "th",
        },
      ],
      "type": "tr",
    },
    "type": "thead",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <tr>
          <th>
                    <SortButtons
                              changeClass={undefined}
                              sort={undefined}
                              sortProperty="name"
                    />
          </th>
          <th
                    className="text-center"
          />
</tr>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <th>
              <SortButtons
                            changeClass={undefined}
                            sort={undefined}
                            sortProperty="name"
              />
</th>,
            <th
              className="text-center"
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <SortButtons
                changeClass={undefined}
                sort={undefined}
                sortProperty="name"
/>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "changeClass": undefined,
                "sort": undefined,
                "sortProperty": "name",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "th",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "text-center",
            },
            "ref": null,
            "rendered": null,
            "type": "th",
          },
        ],
        "type": "tr",
      },
      "type": "thead",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": false,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
